lib/Backend/NativeCodeGenerator.cpp

```diff
@@ -1167,7 +1167,9 @@ NativeCodeGenerator::CheckCodeGen(Js::ScriptFunction * function)
             (
                 originalEntryPoint == DefaultEntryThunk
              || scriptContext->IsDynamicInterpreterThunk(originalEntryPoint)
+#ifdef ENABLE_SCRIPT_PROFILING
```

Comment:
Assert is a macro and VC doesn't like macros within macros. You'll need to ifdef out the whole assert. But more generally, you shouldn't need this here yet since lib/Backend isn't being built on linux.